### PR TITLE
Adds Hover Trait to Actor Item

### PR DESCRIPTION
**Pull request by JoshLaunchcode**
Adds the hover trait to movement.

Reference: #63

### CODE REVIEW

File: src/apps/handlers/prepareMovementData.js [hunk 1]
```diff
@@ -1,10 +1,23 @@
 export default function getMovementData(data) {
   // eslint-disable-next-line no-unused-vars
-  const speeds = Object.entries(data.system.attributes.movement).filter(([_, speed]) => speed);
+  const hover = data.system.attributes.movement.traits.hover
+  const speeds = Object.entries(data.system.attributes.movement).filter(([key, speed]) => {
+    if(!speed && key === "fly" && hover){
+      return !speed
+    } else {
+      return speed
+    }
```
Comment (NekroDarkmoon) [inline src/apps/handlers/prepareMovementData.js:9]:
```js
if ( key === "fly" && !speed && hover) { return true; }
if ( key === "traits") { return false; }
return speed
```

This can be simplified to this.

File: src/apps/handlers/prepareMovementData.js [hunk 2]
```diff
@@ -1,10 +1,23 @@
...
+  }).filter(([key]) => key !== "traits");
```
Comment (NekroDarkmoon) [inline src/apps/handlers/prepareMovementData.js:10]: Using the simplified version above removes the need for the extra filter.

**Review by NekroDarkmoon — CHANGES_REQUESTED** [review]

File: src/apps/handlers/prepareMovementData.js (hunk 1 above)
Comment (Pjb518) [reply src/apps/handlers/prepareMovementData.js]: To stay consistent with our style, this should probably be:
```js
if ( key === "fly" && !speed && hover) return true;
if ( key === "traits") return false;
return speed;
```
That being said, we can do better than this as well. We don't need to check that there's no speed when checking for hover. The presence of the hover trait is enough to show the fly speed regardless of the fly speed value.
```js
if ( key === "fly" && hover) return true;
if ( key === "traits") return false;
return speed;
```

File: public/lang/en.json [hunk 3]
```diff
@@ -487,14 +487,15 @@
...
-  "A5E.MovementHover": "Hover",
   "A5E.MovementConfigurationPrompt": "{name}: Configure Movement Speeds",
   "A5E.MovementBurrowingSpeed": "Burrowing Speed",
   "A5E.MovementBurrowingSpeedAbbr": "Burrowing",
   "A5E.MovementClimbingSpeed": "Climbing Speed",
   "A5E.MovementClimbingSpeedAbbr": "Climbing",
   "A5E.MovementFlyingSpeed": "Flying Speed",
   "A5E.MovementFlyingSpeedAbbr": "Flying",
+  "A5E.MovementHover": "Hover",
+  "A5E.MovementHoverAbbr": "(hover)",
```
Comment (Pjb518) [inline public/lang/en.json:498]: This is not an abbreviation, so we shouldn't be using the `Abbr` suffix for this key. I also don't think we need this key at all. The difference between this key and the one above is formatting.

File: src/apps/handlers/prepareMovementData.js [hunk 4]
```diff
@@ -1,10 +1,21 @@
...
-  const speeds = Object.entries(data.system.attributes.movement).filter(([_, speed]) => speed);
+  const hover = data.system.attributes.movement.traits.hover
+  const speeds = Object.entries(data.system.attributes.movement).filter(([key, speed]) => {
+    if (key === "fly" && !speed && hover) { return true; }
+    if (key === "traits") { return false; }
+    return speed
+  })
+
   const unit = game.i18n.localize('A5E.MeasurementFeetAbbr');
+  const hoverAbbr = game.i18n.localize('A5E.MovementHoverAbbr');
```
Comment (Pjb518) [inline src/apps/handlers/prepareMovementData.js:11]: Replace this with:
```js
hoverLabel = game.i18n.localize('A5E.MovementHover').toLowerCase();
```

File: src/apps/handlers/prepareMovementData.js [hunk 5]
```diff
@@ -1,10 +1,21 @@
...
 
   return speeds.map(([name, speed]) => {
     const label = game.i18n.localize(CONFIG.A5E.movement[name]);
-    return `${label} - ${speed} ${unit}`;
+    if (name == "fly" && hover) {
+      return `${label} - ${speed || 0} ${unit} ${hoverAbbr}`;
```
Comment (Pjb518) [inline src/apps/handlers/prepareMovementData.js:16]: With the change above, we can add the parentheses into the returned string instead:
```js
return `${label} - ${speed || 0} ${unit} (${hoverLabel})`;
```

File: src/apps/handlers/prepareMovementData.js [hunk 6]
```diff
@@ -1,10 +1,21 @@
...
+  const speeds = Object.entries(data.system.attributes.movement).filter(([key, speed]) => {
+    if (key === "fly" && !speed && hover) { return true; }
+    if (key === "traits") { return false; }
+    return speed
+  })
+
   const unit = game.i18n.localize('A5E.MeasurementFeetAbbr');
+  const hoverAbbr = game.i18n.localize('A5E.MovementHoverAbbr');
 
   return speeds.map(([name, speed]) => {
```
Comment (Pjb518) [inline src/apps/handlers/prepareMovementData.js:13]: I don't like the use of `name` here. `movementType` would be more appropriate, I think. Much more explicit.

File: src/apps/handlers/prepareMovementData.js [hunk 7]
```diff
@@ -1,10 +1,21 @@
...
     const label = game.i18n.localize(CONFIG.A5E.movement[name]);
-    return `${label} - ${speed} ${unit}`;
+    if (name == "fly" && hover) {
```
Comment (Pjb518) [inline src/apps/handlers/prepareMovementData.js:15]: I'd like a line break here to separate the conditional block from the variable definitions. Makes it much easier to spot at a glance that these are unrelated pieces of code.

File: src/apps/handlers/prepareMovementData.js (hunk 6 above)
Comment (Pjb518) [reply src/apps/handlers/prepareMovementData.js:13]: It looks like we use `mode` and `distance` in the `MovementConfigDialog`. Regardless of what names we choose, I'd like this to be consistent across these files.

**Review by Pjb518 — CHANGES_REQUESTED** [review]

File: src/apps/handlers/prepareMovementData.js [hunk 8]
```diff
@@ -1,10 +1,21 @@
 export default function getMovementData(data) {
```
Comment (Pjb518) [inline src/apps/handlers/prepareMovementData.js:1]: `data` here is an `actor` so we should use that name to be more explicit. Would appreciate it if you could make that change along with the other stuff 🙂

**Comment by Pjb518** [comment]
@JoshLaunchcode Since we're already updating the movement config in this PR, would you mind switching our movement fields to `number` inputs and ensuring that we save something of type `Number` when updating the actor. May be worth setting a minimum value of `0` for each of the inputs as well.

Thanks 🙂
